core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -174,7 +176,7 @@ DelegateFileIO io(String location) {
     return ioInstances.computeIfAbsent(
         impl,
         key -> {
-          Configuration conf = hadoopConf.get();
+          Configuration conf = getConf();
```

Comment:
I double checked that the CatalogUtil.loadFileIO calls below are null protected as well, the s3fileIO test will exercise that API so that gives confidence too.